core/src/banking_stage/scheduler_messages.rs

```diff
@@ -0,0 +1,76 @@
+use {
+    crate::immutable_deserialized_packet::ImmutableDeserializedPacket,
+    solana_sdk::transaction::SanitizedTransaction, std::sync::Arc,
+};
+
+/// A unique identifier for a transaction batch.
+/// The lower 64 bits are based on the order the transaction batch was scheduled.
+/// The upper 64 bits are an optional priority field - currently unused.
+#[derive(Clone, Copy, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransactionBatchId(u128);
+
+impl TransactionBatchId {
+    pub fn new(index: u64) -> Self {
+        Self(index as u128)
+    }
+}
+
+/// A unique identifier for a transaction.
+/// The lower 64 bits are based on the order the transaction entered banking stage.
+/// The upper 64 bits are the priority of the transaction.
+#[derive(Clone, Copy, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransactionId(u128);
+
+impl TransactionId {
+    pub fn new(priority: u64, index: u64) -> Self {
+        Self(((priority as u128) << 64) | (index as u128))
+    }
+}
+
+/// Message: [Scheduler -> Worker]
+/// Transactions to be consumed (executed, recorded, committed)
+pub struct ConsumeWork {
+    pub batch_id: TransactionBatchId,
+    pub transaction_ids: Vec<TransactionId>,
+    pub transactions: Vec<SanitizedTransaction>,
```

Comment:
these vecs should be limited to size <= 64.

Might consider using a bounded_vec: https://docs.rs/bounded-vec/latest/bounded_vec/struct.BoundedVec.html

Reply:
`64` is a gold standard for now, but it's likely to change with all the improvements are making. Perhaps not too early to set it in stone, or maybe reference it as a const?

Reply:
> 64 is a gold standard for now

not so much that 64 is the standard now, but that it is the maximum number of txs allowed in an entry, right?

Reply:
Taking no action for now, following above conversation